x-pack/legacy/plugins/infra/public/pages/logs/analysis/setup/steps/index.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiSteps, EuiStepStatus } from '@elastic/eui';
+import { InitialConfiguration } from './initial_configuration';
+import { SetupProcess } from './setup_process';
+import { useAnalysisSetupState } from '../../../../../containers/logs/log_analysis/log_analysis_setup_state';
+
+interface AnalysisSetupStepsProps {
+  setup: (startTime?: number | undefined, endTime?: number | undefined) => void;
+  isSettingUp: boolean;
+  didSetupFail: boolean;
+  retry: (startTime?: number | undefined, endTime?: number | undefined) => void;
+  isRetrying: boolean;
+  hasCompletedSetup: boolean;
+  hasAttemptedSetup: boolean;
+  viewResults: () => void;
+  indexPattern: string;
+}
+
+export const AnalysisSetupSteps: React.FunctionComponent<AnalysisSetupStepsProps> = ({
+  setup: setupModule,
+  retry: retrySetup,
+  isSettingUp,
+  didSetupFail,
+  isRetrying,
+  hasCompletedSetup,
+  hasAttemptedSetup,
+  viewResults,
+  indexPattern,
+}: AnalysisSetupStepsProps) => {
+  const { setup, retry, setStartTime, setEndTime, startTime, endTime } = useAnalysisSetupState({
+    setupModule,
+    retrySetup,
+  });
+
+  const steps = [
+    {
+      title: i18n.translate('xpack.infra.analysisSetup.stepOneTitle', {
+        defaultMessage: 'Configuration (optional)',
```

Comment:
Having the optional as a kind of badge or in a smaller font would be great, but I couldn't find a way to achieve that using the `EuiSteps`. :woman_shrugging:

Reply:
Yeah, I couldn't see a way either with `title`s being strict `string`s.